core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -84,10 +84,7 @@ static class DefaultLocationProvider implements LocationProvider {
     private static String dataLocation(Map<String, String> properties, String tableLocation) {
       String dataLocation = properties.get(TableProperties.WRITE_DATA_LOCATION);
       if (dataLocation == null) {
-        dataLocation = properties.get(TableProperties.WRITE_FOLDER_STORAGE_LOCATION);
-        if (dataLocation == null) {
-          dataLocation = String.format("%s/data", tableLocation);
-        }
+        dataLocation = String.format("%s/data", tableLocation);
```

Comment:
Why are we removing public-facing functionality like this? I don't think that we should ever drop support for this because it will break existing tables.

Reply:
I've restored this